src/manual/using/actions/spec.md

```diff
@@ -281,6 +281,39 @@ Note that the result of rendering `hookPayload` becomes the context for another
 JSON-e rendering performed by the hooks service.  Do not confuse the two
 operations!
 
+### Choosing a Kind
+
+The "task" kind requires that the user invoking the task has all of the scopes
+necessary to run the action task (and to create any further tasks). Such scopes
+might allow a user to perform undesirable actions with the Taskcluster API.
+
+The "hook" kind allows more control over users' permissions. The key
+observation is that `hooks.triggerHook` is governed by a scope of the form
+`hooks:trigger-hook:<hookGroupId>/<hookId>`, but the task it creates uses role
+`hook-id:<hookGroupId>/<hookId>`. The method also verifies its payload against
+the hook's `triggerSchema`.  This allows a controlled privilege escalation:
+only clients with the appropriate `trigger-hook` scope may trigger a hook, and
+only if they provide a payload matching the schema. If both of those conditions
```

Comment:
> only if they provide a payload matching the schema

^ this is not a security assertion, as the attacker controls the schema in `actions.json`

Reply:
Maybe rephrase this section... I **love** the sentence "This allows a controlled privilege escalation".

But I'm a little confused about the rest. As an action designer, using a hook means I can grant the user more restricted scopes. And I can use the hook to create a controlled privileged escalation, where my hook implementation sanitizes untrusted input.

Reply:
I meant matching triggerSchema, which the attacker does not control.  What is the confusion?

Reply:
Okay, this is the `triggerSchema` in hooks.. I didn't see that got added.

Curious, will that validate input from both HTTP triggers and pulse triggers...

Reply:
No, triggerSchema only applies to triggers.  I think we decided to add an additional pulseMessageSchema, but I don't recall.  Anyway, that part (pulse) isn't implemented yet.